### PR TITLE
doc: prepare for mpich-4.2.0b1

### LINKED ISSUE #6790: Switch to semantic versioning (SemVer) with explicit patch number even if `0`.
Would you consider switching the version scheme to SemVer for next release? Rather than `4.2b1`, the version would become `4.2.0b1` with an explicit patch entry of `0`.

Many projects have switched to such scheme over the years. Most package managers I know use at least three numbers. The lack of a `0` for the patch version is always a source of annoyance on things like deployment scripts and build recipes ([example](https://github.com/conda-forge/mpich-feedstock/blob/main/recipe/meta.yaml#L5)), which require special-cases and workarounds to cope with jumps in the minor version.

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
Update CHANGE log and mpich version string.

Fixes #6790
[skip warnings]

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -3,6 +3,17 @@
...
+  See paper "Frustrated With MPI+Threads? Try MPIxThreads!",
+  https://doi.org/10.1145/3615318.3615320.
+
+# Experimental datatype functions MPIX_Type_iov_len and MPIX_Type_Iov
+
+# Experimental op MPIX_EQUAL for MPI_Reduce and MPI_Allreduce (intra
+  communicator only)
+
+# CH4 shm is enabled by default with ucx netmod.
+
```
Comment (raffenet) [inline CHANGES:16]:
```suggestion
```

Comment (hzhou) [reply CHANGES:16]: Leaked back in during rebase :( Good catch.

File: maint/version.m4 [hunk 2]
```diff
@@ -14,7 +14,7 @@
 # changing this by playing with diversions, but then we would probably be
 # playing with autotools-fire.
 
-m4_define([MPICH_VERSION_m4],[4.2a1])dnl
+m4_define([MPICH_VERSION_m4],[4.2b1])dnl
```
Comment (raffenet) [inline maint/version.m4:17]:
```suggestion
m4_define([MPICH_VERSION_m4],[4.2.0b1])dnl
```

File: CHANGES [hunk 3]
```diff
@@ -22,6 +31,28 @@
 # Yaksa updated to auto detect the GPU architecture and only build for
   the detected arch. This applies to CUDA and HIP support.
 
+# MPI_Win_shared_query can be used on windows created by MPI_Comm_create,
+  MPI_Comm_allocate, in addition to windows created by MPI_Comm_allocate_shared.
```
Comment (raffenet) [inline CHANGES:35]:
```suggestion
# MPI_Win_shared_query can be used on windows created by MPI_Win_create,
  MPI_Win_allocate, in addition to windows created by MPI_Win_allocate_shared.
```

File: CHANGES [hunk 4]
```diff
@@ -3,6 +3,15 @@
 ===============================================================================
 # Complete support MPI 4.1 specification
 
+# Experimental threadd communicator feature (e.g. MPIX_Threadcomm_init).
```
[the remark this answers is not in the document]
Comment (hzhou) [reply CHANGES]: Good catch! Fixed.

File: CHANGES (hunk 3 above)
Comment (hzhou) [reply CHANGES]: Good catch! Fixed.

**Review by raffenet — APPROVED** [review]

**Pull request merged**